uniswap_swap_detection/src/agent.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} from "forta-agent";
+
+import { SWAP_EVENT } from "./constant";
+import { poolABI } from "./abi";
+
+import { uniswapPoolAddress } from "./utils";
+
+
+const PROVIDER = getEthersProvider();
+const block = PROVIDER.getBlockNumber();
```

Comment:
This is an async function and as it's not awaited it returns a Promise.

For the `blockTag`, `txEvent.blockNumber`should be used instead. Please make this change in the code and tests and I'll then provide a full review.